tests/proxy/map_paging_predicate_test.py

```diff
@@ -0,0 +1,241 @@
+
+        def assert_event():
+            assertCountEqual(self, self.map.values(paging).result(), [0, 0, 1, 1, 2])
+            paging.next_page()
+            assertCountEqual(self, self.map.values(paging).result(), [2, 3, 3, 4, 4])
+            paging.next_page()
+            assertCountEqual(self, self.map.values(paging).result(), [5, 5, 6, 6, 7])
+            paging.next_page()
+            assertCountEqual(self, self.map.values(paging).result(), [7, 8, 8])
+        self.assertTrueEventually(assert_event)
+
+    """
+    Test for paging predicate with custom comparator
+    """
+    def test_key_set_paging_with_custom_comparator(self):
+        self._fill_map_custom_comp()
+        custom_cmp = CustomComparator(type=1, iteration_type=ITERATION_TYPE.KEY)
+        paging = PagingPredicate(is_ilike('__key', 'key-%'), 6, custom_cmp)
+
+        key_set_page_1 = self.map.key_set(paging).result()
+        paging.next_page()
+        key_set_page_2 = self.map.key_set(paging).result()
+        paging.next_page()
+        key_set_page_3 = self.map.key_set(paging).result()
+
+        def assert_event():
+            assertCountEqual(self, key_set_page_1, ['key-9', 'key-8', 'key-7', 'key-6', 'key-5', 'key-4'])
+            assertCountEqual(self, key_set_page_2, ['key-3', 'key-2', 'key-1', 'key-0'])
+            assertCountEqual(self, key_set_page_3, [])
+
+        self.assertTrueEventually(assert_event)
+
+    def test_values_paging_with_custom_comparator(self):
+        self._fill_map_custom_comp_2()
+        custom_cmp = CustomComparator(type=2, iteration_type=ITERATION_TYPE.VALUE)
+        paging = PagingPredicate(None, 6, custom_cmp)
+
+        values_page_1 = self.map.values(paging).result()
+        paging.next_page()
+        values_page_2 = self.map.values(paging).result()
+        paging.next_page()
+        values_page_3 = self.map.values(paging).result()
+
+        def assert_event():
+            assertCountEqual(self, values_page_1, ['A', 'BB', 'CCC', 'DDDD', 'EEEEE', 'FFFFFF'])
+            assertCountEqual(self, values_page_2, ['GGGGGGG', 'HHHHHHHH', 'IIIIIIIII', 'JJJJJJJJJJ'])
+            assertCountEqual(self, values_page_3, [])
+
+        self.assertTrueEventually(assert_event)
+
+    def test_entry_set_paging_with_custom_comparator(self):
+        self._fill_map_custom_comp_2()
+        custom_cmp = CustomComparator(type=2, iteration_type=ITERATION_TYPE.ENTRY)
+        paging = PagingPredicate(None, 2, custom_cmp)
+
+        page1 = self.map.entry_set(paging).result()
+        self.assertTrueEventually(lambda: assertCountEqual(self, page1, [('key-65', 'A'), ('key-66', 'BB')]))
+
+    def _fill_map_simple(self):
+        self.map.put('key-1', 1)
+        self.map.put('key-2', 2)
+        self.map.put('key-3', 3)
+
+    def _fill_map_numeric(self, count=50):
+        for i in range(count):
+            self.map.put(i, i)
+
+    def _fill_map_custom_comp(self, count=10):
+        for i in range(count):
```

Comment:
This can be replaced with two API calls (or with one if you are willing to sacrifice some readability using dict comprehension, zip and some itertools functionality)